sys/include/net/dhcpv6/client.h

```diff
@@ -170,6 +198,46 @@ void dhcpv6_client_conf_prefix(unsigned netif, const ipv6_addr_t *pfx,
                                unsigned pfx_len, uint32_t valid,
                                uint32_t pref);
 
+/**
+ * @brief   Checks if the given network interface is configured
+ *          to use DHCPv6 IA_NA
+ *
+ * @param[in] netif     Network interface to check.
+ *
+ * @return  true, if the network interface is set up for IA_NA.
+ */
+bool dhcpv6_client_check_ia_na(unsigned netif);
+
+/**
+ * @brief   Configures a address lease that is provided by the server.
+ *
+ * @param[in] netif     Network interface the address was for.
+ * @param[in] addr      The assigned address.
+ *
+ * @return  true, if the address was configured successfully.
+ */
+bool dhcpv6_client_add_addr(unsigned netif, ipv6_addr_t *addr);
+
+/**
+ * @brief   Deprecates an existing address from an address lease.
+ *
+ * @param[in] netif     Network interface the address was for.
+ * @param[in] addr      The address to deprecate.
+ *
+ * @return  true, if the address was deprecated successfully.
+ */
+bool dhcpv6_client_deprecate_addr(unsigned netif, const ipv6_addr_t *addr);
+
+/**
+ * @brief   Removes an existing address that originated from an address lease.
+ *
+ * @param[in] netif     Network interface the address was for.
+ * @param[in] addr      The address to remove.
+ *
+ * @return  true, if the address was removed successfully.
+ */
+bool dhcpv6_client_remove_addr(unsigned netif, ipv6_addr_t *addr);
```

Comment:
Here too.

Reply:
The signature of `gnrc_netif_ipv6_addr_remove` (and `gnrc_netif_ipv6_addr_add`, respectively) is 

```c
static inline int gnrc_netif_ipv6_addr_remove(const gnrc_netif_t *netif, ipv6_addr_t *addr)
```

which lead to the compiler error 

> error: passing argument 2 of ‘gnrc_netif_ipv6_addr_add’ discards ‘const’ 
> qualifier from pointer target type [-Werror=discarded-qualifiers]

This was also the reason why I used the internal functions as those actually have `const` address parameters. Is there an easy fix for this problem or would the parameters of the gnrc functions have to be made `const` for this as well?

Reply:
Ah, this is because it is used with `gnrc_netapi_set()` where in turn it is used to set a pointer in an `msg_t`. So even though it _could_ be used as `const` (the `NETAPI_SET` routine of `gnrc_netif` treats it as an input pointer), doing so would probably still through some warnings in the compiler, even when explicitly casted, depending on the level of warnings. So I agree to leave it as is.